main.py

```diff
@@ -17,9 +20,50 @@ class Dishes(Resource):
     def get(self):
         return ResponseSerializer(col.get_dishes(), 200).serialize()
 
-
     def delete(self):
         return ResponseSerializer({ }, 405).serialize()
+    
+    def post(self):
+        # Only accept app/json content type
+        if request.content_type != 'application/json':
+            return ResponseSerializer(0, 415).serialize()
+        
+        req_json = request.get_json()
+
+        validator = DishValidator(req_json, 'post').call()
+
+        if not validator.valid:
+            return ResponseSerializer(-1, 422).serialize()
+        
+        # Check if a dish with that name already exists
+        if col.find_data_item(col.dishes, 'name', req_json['name']) != -1:
+            return ResponseSerializer(-2, 422).serialize()
+
+        # Calculate nutrition from API-ninjas
+        query = req_json['name']
+        api_url = 'https://api.api-ninjas.com/v1/nutrition?query={}'.format(query)
+
+        response = requests.get(api_url, headers={'X-Api-Key': 'xGz7WUhOfPLpoGoeKxEO+w==5cBTcX74GCXh7AHX'})
```

Comment:
not good practice to have api key here.

Let's create a file with the name `.env`, and add it to the gitignore.
Then, in that file, we give it a name like:
`API_NINJA_KEY=xGz7WUhOfPLpoGoeKxEO+w==5cBTcX74GCXh7AHX`

and then we can do 

```        
...
import os
...

response = requests.get(api_url, headers={'X-Api-Key': os.environ.get("API_NINJA_KEY") })
```

More secure way to going about it. can each add this file individually as it won't be tracked on git

Reply:
put in different file and export